src/client/builtin_scene/instanced_mesh.cpp

```diff
@@ -548,9 +554,62 @@ namespace builtin_scene
       return;
 
     shared_lock<shared_mutex> lock(mutex_);
+
+    // Update layered instances based on RenderLayer from instances
+    std::map<RenderLayer, InstanceMap> layeredInstanceMaps;
+    for (auto &[id, instance] : idToInstanceMap_)
+    {
+      if (instance != nullptr && !instance->skipToDraw())
+      {
+        RenderLayer layer = instance->renderLayer_;
+        layeredInstanceMaps[layer][id] = instance;
+      }
+    }
+
+    // Update or create RenderableInstancesList for each layer
+    for (auto &[layer, instanceMap] : layeredInstanceMaps)
+    {
+      if (layeredInstances_.find(layer) == layeredInstances_.end())
+      {
+        // Create new RenderableInstancesList for this layer with dedicated VAO and VBO
+        auto glContext = glContext_.lock();
+        if (glContext != nullptr)
+        {
+          auto layerVao = glContext->createVertexArray();
+          auto layerInstanceVbo = glContext->createBuffer();
+          layeredInstances_[layer] = std::make_shared<RenderableInstancesList>(
```

Comment:
You can configure the vertex pointers for the instance vbo by moving `iterateInstanceAttributes` into `RenderableInstancesList` constructor.